src/DiagramGenerator/Config/Texture.php

```diff
@@ -11,34 +12,75 @@
  */
 class Texture
 {
+    /** @var string $name */
+    protected $name;
+
+    /** @var string $imageUrlFolderName */
+    protected $imageUrlFolderName;
+
+    /** @var string $imageFormat */
+    protected $imageFormat;
+
+    /** @var string $highlightSquaresColor The default highlight squares color for the board texture */
+    protected $highlightSquaresColor;
+
     /**
-     * Boards filename
-     * @Type("string")
-     * @var string
+     * @param string      $name
+     * @param string      $imageUrlFolderName
+     * @param string      $imageFormat
+     * @param string|null $highlightSquaresColor
      */
-    protected $board;
+    public function __construct($name, $imageUrlFolderName, $imageFormat, $highlightSquaresColor = null)
+    {
+        if (!in_array($imageFormat, array('png', 'jpg'))) {
+            throw new InvalidArgumentException(sprintf('Invalid image format: %s', $imageFormat));
+        }
+
+        $this->name = $name;
+        $this->imageUrlFolderName = $imageUrlFolderName;
+        $this->imageFormat = $imageFormat;
+        $this->highlightSquaresColor = $highlightSquaresColor;
+    }
 
     /**
-     * Gets the Boards filename.
-     *
      * @return string
      */
-    public function getBoard()
+    public function getName()
     {
-        return $this->board;
+        return $this->name;
     }
 
     /**
-     * Sets the Boards filename.
-     *
-     * @param string $board the board
-     *
-     * @return self
+     * @return string
+     */
+    public function getImageUrlFolderName()
+    {
+        return $this->imageUrlFolderName;
+    }
+
+    /**
+     * @return string
+     */
+    public function getImageFormat()
+    {
+        return $this->imageFormat;
+    }
+
+    /**
+     * @return string
      */
-    public function setBoard($board)
+    public function getHighlightSquaresColor()
     {
-        $this->board = $board;
+        return $this->highlightSquaresColor;
+    }
 
-        return $this;
+    /**
+     * @param DiagramGenerator\Config\Texture $texture
+     *
+     * @return boolean
+     */
+    public function equals(Texture $texture)
```

Comment:
We have similar method in `User` entity ([is() method](https://github.com/ChessCom/chess/blob/develop/src/Chess/WebBundle/Entity/User.php#L3623)). Maybe you should name this one the same.